src/core/path.c

```diff
@@ -53,8 +54,15 @@ QuicPathRemove(
     _In_ uint8_t Index
     )
 {
-    CXPLAT_DBG_ASSERT(Index < Connection->PathsCount);
+    CXPLAT_DBG_ASSERT(Connection->PathsCount > 0);
+    CXPLAT_DBG_ASSERT(Connection->PathsCount <= QUIC_MAX_PATH_COUNT);
```

Comment:
why don't we make this a teleassert too?

Reply:
We _could_ but I really don't expect this assert to be invalid. I am not so sure about the next one, that's why I've made it a tel assert.